src/sdam/topology.rs

```diff
@@ -477,23 +477,25 @@ impl TopologyWorker {
         let old_description = latest_state.description.clone();
 
         if let Some(expected_name) = &self.options.repl_set_name {
-            let got_name = sd.set_name();
-            if latest_state.description.topology_type() == TopologyType::Single
-                && got_name.as_ref().map(|opt| opt.as_ref()) != Ok(Some(expected_name))
-            {
-                let got_display = match got_name {
-                    Ok(Some(s)) => format!("{:?}", s),
-                    Ok(None) => "<none>".to_string(),
-                    Err(s) => format!("<error: {}>", s),
-                };
-                // Mark server as unknown.
-                sd = ServerDescription::new(
-                    sd.address,
-                    Some(Err(format!(
-                        "Connection string replicaSet name {:?} does not match actual name {}",
-                        expected_name, got_display,
-                    ))),
-                );
+            if sd.is_available() {
```

Comment:
I updated this block to only run when the server being updated is available, since it was running any time a server description with an error was being used to update (error => no set name).